model/device_deployment.go

```diff
@@ -353,8 +354,31 @@ func InactiveDeploymentStatuses() []DeviceDeploymentStatus {
 // InstalledDeviceDeployment describes a deployment currently installed on the
 // device, usually reported by a device
 type InstalledDeviceDeployment struct {
-	ArtifactName string `json:"artifact_name"`
-	DeviceType   string `json:"device_type"`
+	ArtifactName string            `json:"artifact_name"`
+	DeviceType   string            `json:"device_type"`
+	Provides     map[string]string `json:"artifact_provides,omitempty"`
+}
+
+// DeploymentNextRequest holds a deployments/next request
+type DeploymentNextRequest struct {
+	DeviceProvides   *InstalledDeviceDeployment `json:"device_provides"`
+	UpdateControlMap bool                       `json:"update_control_map"`
+}
+
+func (i *DeploymentNextRequest) Validate() error {
+	return validation.ValidateStruct(i,
+		validation.Field(&i.DeviceProvides,
+			validation.Required,
+		),
+	)
+}
+
+func (i *DeploymentNextRequest) String() string {
```

Comment:
Q: What do we need the `fmt.Stringer` interface for?

Reply:
only to nicely log old and new request when they are different